src/scripts/main.js

```diff
@@ -1,3 +1,243 @@
+      const itemB = stringOrNumber(b.children[i].textContent);
+
+      return typeof itemA === 'number'
+        ? itemA - itemB
+        : itemA.localeCompare(itemB);
+    });
+
+    list.append(...arr);
+  } else {
+    sortReverse = false;
+
+    arr.sort((a, b) => {
+      const itemA = stringOrNumber(a.children[i].textContent);
+      const itemB = stringOrNumber(b.children[i].textContent);
+
+      return typeof itemA === 'number'
+        ? itemB - itemA
+        : itemB.localeCompare(itemA);
+    });
+
+    list.append(...arr);
+  }
+}
+
+function stringOrNumber(string) {
+  const DIGITS = '0123456789';
+  let str = '';
+  let num = '';
+
+  string.split('').map(char => {
+    DIGITS.includes(char)
+      ? num += char
+      : str += char;
+  });
+
+  return str.length > num.length ? str : +num;
+}
+
+// ------------------ACTIVE ROW------------------
+
+list.addEventListener('click', (e) => {
+  const active = [...document.querySelectorAll('.active')];
+
+  const item = e.target.closest('tr');
+
+  item.classList.toggle('active');
+
+  if (active.length > 0) {
+    active.map((el) => el.classList.remove('active'));
+  }
+});
+
+// -------------------CREATING FORM-----------------
+
+document.body.insertAdjacentHTML('beforeend', `
+  <form action="" class='new-employee-form'>
+    <label>
+      Name:
+      <input name="name" type="text" data-qa="name">
+    </label>
+
+    <label>
+      Position:
+      <input name="position" type="text" data-qa="position">
+    </label>
+
+    <label>
+      Office:
+      <select name="office" data-qa="office">
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+
+    <label>
+      Age:
+      <input name="age" type="number" data-qa="age" min="0">
+    </label>
+
+    <label>
+      Salary:
+      <input name="salary" type="number" data-qa="salary" min="0">
+    </label>
+
+    <button type="submit">
+      Save to table
+    </button>
+  </form>
+`);
+
+// -------------ADDING A NEW PERSON------------------
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const data = new FormData(form);
+  const values = Object.fromEntries(data);
+
+  if (values.name.length < 4) {
+    pushNotification('error', 'Short name. Min length 4', 'error');
+
+    return;
+  };
+
+  if (+values.age < 18 || +values.age > 90) {
+    pushNotification('error', 'Invalid age. Range 18-80', 'error');
+
+    return;
+  }
+
+  let isFieldsFilled = false;
+
+  for (const key in values) {
+    if (values[key] === '') {
+      isFieldsFilled = true;
+      break;
+    }
+  }
+
+  if (isFieldsFilled) {
+    pushNotification('error', 'Fill in all the fields', 'error');
+
+    return;
+  } else {
+    pushNotification('success', 'New person added', 'success');
+  }
```

Comment:
you don't need an `else` if you do return in the `if` block
```suggestion
  } 
  
  pushNotification('success', 'New person added', 'success');
  
```